backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/address/AddressDescriptorFactory.java

```diff
@@ -108,17 +125,25 @@ private DocumentEntityDescriptor createAddressEntityDescriptor()
 				.setValueClass(String.class)
 				.setWidgetType(DocumentFieldWidgetType.Text)
 				.setDataBinding(new AddressFieldBinding(IAddressModel.COLUMNNAME_Address2, false, I_C_Location::getAddress2, AddressFieldBinding::writeValue_Address2)));
-		//
-		addressDescriptor.addField(buildFieldDescriptor(IAddressModel.COLUMNNAME_Address3)
-				.setValueClass(String.class)
-				.setWidgetType(DocumentFieldWidgetType.Text)
-				.setDataBinding(new AddressFieldBinding(IAddressModel.COLUMNNAME_Address3, false, I_C_Location::getAddress3, AddressFieldBinding::writeValue_Address3)));
-		//
-		addressDescriptor.addField(buildFieldDescriptor(IAddressModel.COLUMNNAME_Address4)
-				.setValueClass(String.class)
-				.setWidgetType(DocumentFieldWidgetType.Text)
-				.setDataBinding(new AddressFieldBinding(IAddressModel.COLUMNNAME_Address4, false, I_C_Location::getAddress4, AddressFieldBinding::writeValue_Address4)));
-		//
+
+		final boolean allowAddress3 = isShowAddress3();
+		if(allowAddress3)
```

Comment:
cosmetics: 

just inline this. there is no need for a local var when it's the only place where is used.

i.e. change it to
```
if(isShowAddress3()) { 
....

```

Reply:
Done